mbuild/formats/lammpsdata.py

```diff
@@ -927,7 +992,10 @@ def _get_dihedral_types(
         structure.join_dihedrals()
         for dihedral in structure.dihedrals:
             if not dihedral.improper:
-                weight = 1 / len(dihedral.type)
+                if zero_dihedral_weighting_factor:
+                    weight = 0.0
+                else:
+                    weight = 1.0 / len(dihedral.type)
```

Comment:
I may have missed it, but should we make a test case that covers when this is not 1 or just set it to 1 if it will never be not 1.?

Reply:
If anything, we should verify that the assert listed in the tests validate that the weighting factor is being applied like we expect.

Reply:
@bc118 This PR only really covers support for AMBER force fields, and will maintain the prior (incorrect) determination of the weighting factor for CHARMM force fields because they are not AMBER.  The addition of `zero_dihedral_weighting_factor` ensures that the LAMMPS data writer can handle AMBER proper dihedrals correctly.  As this is written, the weighting factor can take values other than 1 in the case of multi-term dihedrals when `zero_dihedral_weighting_factor = False`.

I intend to follow the suggestion from @justinGilmer by including assertions that the weighting factor is zero when `zero_dihedral_weighting_factor = True` and the weighting factor is 1 for a single-term dihedral when `zero_dihedral_weighting_factor = False` (choosing the dihedral such that this behavior should not change when the code is eventually updated to determine CHARMM dihedral weighting factors correctly).

Reply:
The first assertion was already there and I just added the other one.